server/text_generation_server/models/custom_modeling/flash_santacoder_modeling.py

```diff
@@ -464,8 +464,9 @@ def forward(
 
 
 class FlashSantacoderForCausalLM(nn.Module):
-    def __init__(self, config, weights):
+    def __init__(self, prefix, config, weights):
         super().__init__()
+        config.transpose = config.architectures[0].startswith("GPT2")
         self.transformer = FlashSantacoderModel(config, weights)
```

Comment:
I think this will also need to be `self.model` instead, otherwise the iteration in `adapter_target_to_layer` from `flash_causal_lm.py` does not work.

Reply:
I don' think loras for Santacoder even exist, right ?

Even if that's the case, ideally we want to push the logic about layer loads into the model itself (which makes more sense than keeping around random names in `flash_causal_lm.py`.

Reply:
Yeah, indeed seems like a better place than `flash_causal_lm.py`.